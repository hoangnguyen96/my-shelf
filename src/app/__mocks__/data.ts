export const DATA_BOOKS = [
  {
    title: "Don't Make Me Think",
    author: "Steve Krug",
    category: "SelfHelpBook",
    imageUrl: "https://i.ibb.co/QbQgVtG/book1.png",
    description: "description will update later",
    status: false,
    publicationYear: 2000,
    rating: 4.5,
    createdDate: "3 Sep 2024 9:07 PM",
    edition: "first",
    id: "1",
  },
  {
    title: "Rich Dad Poor Dad",
    author: "Robert T. Kiyosaki",
    category: "SelfHelpBook",
    imageUrl: "https://i.ibb.co/9qHTtQF/book2.png",
    description:
      "Steve Krug is a usability consultant who has more than 30 years of experience as a user advocate for companies like Apple, Netscape, AOL, Lexus, and others. Based in part on the success of his first book, Don't Make Me Think, he has become a highly sought-after speaker on usability design.",
    status: false,
    publicationYear: 1997,
    rating: 4.5,
    createdDate: "3 Sep 2024 8:47 PM",
    edition: "second",
    id: "2",
  },
  {
    title: "The Design of Everyday Things",
    author: "Don Norman",
    category: "TextBook",
    imageUrl: "https://i.ibb.co/LnSBcCM/book5.png",
    description:
      "Steve Krug is a usability consultant who has more than 30 years of experience as a user advocate for companies like Apple, Netscape, AOL, Lexus, and others. Based in part on the success of his first book, Don't Make Me Think, he has become a highly sought-after speaker on usability design.",
    status: false,
    publicationYear: 1988,
    rating: 4.5,
    createdDate: "3 Sep 2024 8:51 PM",
    edition: "fourth",
    id: "3",
  },
];

export const DATA_USER = [
  {
    username: "admin",
    email: "admin@gmail.com",
    password: "$2a$10$MwSZ9vWSX7DVJwJ4SF5Mzuj2zMFjY52XM9L11gGG0j79gybowaLfq",
    isAdmin: true,
    phone: "",
    bio: "",
    avatar: "https://i.ibb.co/RHMqQGr/man-1.png",
    favorites: ["3", "2", "1", "13", "5"],
    shelfBooks: ["1", "6"],
    userId: "3733403",
    id: "1",
  },
  {
    username: "hung",
    email: "hung1@gmail.com",
    password: "$2a$10$Mt6ruwOzt3nzX1.HpSt.OOzut.teXmLE74WclmkP.Nn.gzVQrlqri",
    isAdmin: false,
    phone: "",
    bio: "",
    avatar: "https://i.ibb.co/88X1WfZ/avatar-default.png",
    favorites: ["1"],
    shelfBooks: ["2", "3", "1"],
    userId: "8857134",
    id: "2",
  },
];
