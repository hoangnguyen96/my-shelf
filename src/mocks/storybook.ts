export const mockRouter = {
  pathname: "/",
  route: "/",
  query: {},
  asPath: "/",
  basePath: "",
  push: () => Promise.resolve(true),
  replace: () => Promise.resolve(true),
  reload: () => {},
  back: () => {},
  prefetch: () => Promise.resolve(),
  beforePopState: () => {},
  isFallback: false,
  isLocaleDomain: false,
  isReady: true,
  isPreview: false,
  forward: () => {},
  events: {
    on: () => {},
    off: () => {},
    emit: () => {},
  },
};

export const mockSession = {
  user: {
    isAdmin: true,
    email: "admin@gmail.com",
    id: "3733403",
    name: "admin",
    image: "https://i.ibb.co/RHMqQGr/man-1.png",
  },
  expires: "2024-12-31T23:59:59.999Z",
};
