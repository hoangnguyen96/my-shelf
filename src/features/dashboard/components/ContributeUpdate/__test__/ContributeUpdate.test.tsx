import { render } from "@testing-library/react";
import { DATA_BOOKS } from "@app/mocks/data";
import { ContributeUpdate } from "..";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

describe("Contribute Update", () => {
  it("Should render correctly snapshot", () => {
    const { container } = render(<ContributeUpdate book={DATA_BOOKS[0]} />);

    expect(container).toMatchSnapshot();
  });
});
