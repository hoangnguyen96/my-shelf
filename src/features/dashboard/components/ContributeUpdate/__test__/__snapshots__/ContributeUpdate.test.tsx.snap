// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contribute Update Should render correctly snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <p
      class="chakra-text css-1jr2z3f"
    >
      Preview Book Details
    </p>
    <form
      class="css-79elbk"
    >
      <div
        class="css-197p3oi"
      >
        <div
          class="css-1sfiv63"
        >
          <div
            class="chakra-form-control css-5b2hom"
            role="group"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <input
                class="chakra-input css-1x36n8t"
                id="title"
                name="title"
                placeholder="Book name"
                type="text"
                value="Don't Make Me Think"
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-5b2hom"
            role="group"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <input
                class="chakra-input css-1x36n8t"
                id="author"
                name="author"
                placeholder="Author Name"
                type="text"
                value="Steve Krug"
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-7m4uk4"
            role="group"
          >
            <textarea
              class="chakra-textarea css-1yn82jt"
              id="field-:r2:"
              name="description"
              placeholder="Reason For Your Contribution"
            >
              description will update later
            </textarea>
          </div>
        </div>
        <div
          class="css-j7qwjs"
        >
          <div
            class="chakra-form-control css-106bepj"
            role="group"
          >
            <div
              class="chakra-select__wrapper css-42b2qy"
            >
              <select
                aria-label="select-category"
                class="chakra-select css-g3rz38"
                data-testid="contribute-category"
                id="field-:r3:"
              >
                <option
                  value=""
                >
                  Category
                </option>
                <option
                  class="css-0"
                  value="SelfHelpBook"
                >
                  SelfHelpBook
                </option>
                <option
                  class="css-0"
                  value="TextBook"
                >
                  TextBook
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-162mkon"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="chakra-form-control css-1xz9bny"
            role="group"
          >
            <label
              class="chakra-form__label css-zfpjyo"
              data-testid="fileInput"
              for="fileInput"
              id="field-:r4:-label"
            >
              <img
                alt="Upload Image"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="100"
                src="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FQbQgVtG%2Fbook1.png&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fi.ibb.co%2FQbQgVtG%2Fbook1.png&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fi.ibb.co%2FQbQgVtG%2Fbook1.png&w=256&q=75 2x"
                style="color: transparent; margin: 0px auto;"
                width="100"
              />
            </label>
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <input
                accept="image/*"
                class="chakra-input css-1vuqmsh"
                data-testid="input-file-contribute"
                id="fileInput"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
      <button
        class="chakra-button css-1qhatj0"
        data-testid="submit-contribute"
        disabled=""
        type="submit"
      >
        Update
      </button>
    </form>
  </div>
</div>
`;
